scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+
+import argparse
+import shlex
+import subprocess
+import pathlib
+import platform
+import sys
+import shutil
+import multiprocessing
+import os
+
+# Resolve SOF_TOP directory as: "this script directory/.."
+SOF_TOP=pathlib.Path(__file__).parents[1].resolve()
+WEST_TOP=pathlib.Path(SOF_TOP, "zephyrproject")
+default_rimage_key=pathlib.Path("modules", "audio", "sof", "keys", "otc_private_key.pem")
+
+if platform.system() == "Windows":
+	xtensa_tools_version_postfix = "-win32"
+elif platform.system() == "Linux":
+	xtensa_tools_version_postfix = "-linux"
+else:
+	raise RuntimeError(f"Your operating system: {platform.system()} is not supported")
+
+platform_list = [
```

Comment:
Please keep the distinction between default (`-a`) platforms and the superset of supported platforms. It's not used right now but it's been useful in the past and it's a small code change, just two variables instead of one.

Reply:
@marc-hb please elaborate I do not understand what do you mean.
Currently `-a` adds all platform names from `platform_list` to build queue. I though we support all platforms that are on that list?

Reply:
There are times when a platform works for only certain people who have the "special sauce" required: unusual dependencies, bleeding edge toolchain, temporary workarounds, etc. There has also been times when we temporarily "disabled" some troublesome platform. That's why the shell script has two tiers:
- The `-a` default platforms that everyone including CI can and should build whenever they make a change.
- The other, "in progress" or "more difficult" additional platforms.

Reply:
If I understand you correctly, you mean to change just the naming - from "all_platforms" to "supported_platforms" correct? OK done.